tests/vcs.py

```diff
@@ -384,3 +379,60 @@ def test_stable_branch(repo, monkeypatch):
         merge_msg,
         repo.user_full,
     ]
+
+
+def test_vcs_deploy_with_branchname(repo, monkeypatch):
+    if repo.vcs_type == 'git':
+        branch_name = 'featureXXX'
+
+        # Make another commit
+        repo.store_commit_hash('7.txt', branch=branch_name)
+
+        # test that it was stored
+        assert repo.commit_hash.get('7.txt', None)
+
+        # Push to remote
+        repo.push(branch=branch_name)
+
+        # set host string
+        monkeypatch.setattr('fabric.state.env.host_string', 'staging.hammer')
+        monkeypatch.setattr('fabric.state.env.use_ssh_config', True)
+
+        # Run deploy
+        obj = repo.get_vcs()
+
+        # Run deploy with branch name
+        obj.update(revision=branch_name)
+
+        # The version info should be correct
+        assert list(obj.version()) == [
+            repo.commit_hash['7.txt'],
+            branch_name,
+            '7.txt',
+            repo.user_full,
+        ]
+
+        # Also test that we get correct branch
+        assert obj.get_branch() == branch_name
```

Comment:
Can you guys also add a test which:

1. creates branch `top_secret` and adds a single commit there
2. adds 1 more commit to master
3. executes `deployment_list` on the remote and verifies that `deployment_list` returns commit created in step 2
4. executes `deployment_list("top_secret")` and verifies it's output is correct
5. executes `update("top_secret")` and verifies the resulting version is correct